ext/websocket/01_websocket.js

```diff
@@ -446,6 +456,41 @@
         }
       }
     }
+
+    [_serverHandleIdleTimeout]() {
+      if (this[_idleTimeoutDuration]) {
+        clearTimeout(this[_idleTimeoutTimeout]);
+        this[_idleTimeoutTimeout] = setTimeout(async () => {
+          await core.opAsync("op_ws_send", this[_rid], {
+            kind: "ping",
+          });
+          this[_idleTimeoutTimeout] = setTimeout(() => {
+            this[_readyState] = CLOSING;
+            PromisePrototypeThen(
+              core.opAsync("op_ws_close", {
+                rid: this[_rid],
+                code,
+                reason,
+              }),
+              () => {
+                this[_readyState] = CLOSED;
+
+                const errEvent = new ErrorEvent("error");
```

Comment:
Do error events not have error messages?

Reply:
I brought this up months ago and then completely forgot about it, but the `error` event thrown by the `WebSocket` API is an instance of `Event`, not of `ErrorEvent`. Step 2 in ["when the WebSocket connection is closed"](https://websockets.spec.whatwg.org/#closeWebSocket) throws an event named `error`, not an `ErrorEvent`.

Reply:
But I guess this doesn't have to block this PR.

Reply:
I'd find that kinda counter-productive since `Event` doesnt allow having a message, unlike ErrorEvent. i'd rather break spec for this than have no clue why a websocket is erroring